rktl_game/nodes/game_manager

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+"""Use ball location to determine when a goal is scored.
+License:
+  BSD 3-Clause License
+  Copyright (c) 2020, Autonomous Robotics Club of Purdue (Purdue ARC)
+  All rights reserved.
+"""
+
+
+
+# ROS
+import rospy
+from std_msgs.msg import Bool, UInt16, Int16, Int8, Header
+from std_srvs.srv import Empty, EmptyResponse, Trigger, TriggerResponse
+from nav_msgs.msg import Odometry
+from rktl_msgs.msg import Score, MatchStatus
+
+class ScoreKeeper():
+    def __init__(self):
+        rospy.init_node('score_keeper')
+
+        self.orange_score = 0
+        self.blue_score = 0
+        self.orange_goal = (rospy.get_param('/field/length', 1) + 0.1) / 2 
+        self.blue_goal = (rospy.get_param('/field/length', 1)  + 0.1) / -2
+        self.enabled = False
+        self.match_status = 0
+        self.game_clock = rospy.get_param('/game_length', 90)
+
+        # Publishers
+        self.match_status_pub = rospy.Publisher('match_status', MatchStatus, queue_size=1)
+        self.active_pub = rospy.Publisher('cars/enable', Bool, queue_size=1)
+
+        # Subscribers
+        rospy.Subscriber('ball/odom', Odometry, self.check_goal)
+        rospy.Subscriber('cars/enable', Bool, self.enable)
+
+        # Services
+        rospy.Service('reset_game', Empty, self.reset)
+        rospy.Service('unpause_game', Empty, self.unpause)
+        rospy.Service('pause_game', Empty, self.pause)
+
+        # Counts loops and decrements game clock every second
+        self.timer_ctr = 0
+        # self.unpause(self)
+        # main loop
+        rate = rospy.Rate(10) #10hz
+        while not rospy.is_shutdown():
+            if self.match_status == 1:
+                self.timer_ctr += 1
+                if self.timer_ctr == 10:
+                    self.timer_ctr = 0
+                    self.game_clock -= 1
+            self.update_status()
+            rate.sleep()
+
+    def update_status(self):
+        if (self.match_status == 1 and self.game_clock == 0):
+            if (self.orange_score > self.blue_score):
+                self.match_status = 3
+            else:
+                self.match_status = 4
+        score = Score(self.orange_score, self.blue_score)
+        self.match_status_pub.publish(MatchStatus(self.match_status, self.game_clock, score))
+
+    def pause(self, _):
+        self.enabled = False
+        self.match_status = 0
+        self.active_pub.publish(False)
+        return EmptyResponse()
+
+    def reset(self, _):
+        self.orange_score = 0
+        self.blue_score = 0
+        self.enabled = False
+        self.match_status = 0
+        self.game_clock = rospy.get_param('/game_length', 90)
```

Comment:
Since only your node consumes this info, I think it can be a private param

Reply:
The webpage uses it